pkg/didcomm/protocol/didexchange/persistence_test.go

```diff
@@ -121,24 +121,105 @@ func TestConnectionRecorder_GetInvitation(t *testing.T) {
+			ConnectionID: "connID", State: stateNameInvited}
+		err := record.saveConnectionRecord(connRec)
 		require.NoError(t, err)
-		require.Equal(t, "value1", v.State)
+		v, err := record.GetConnectionRecord("xyz")
+		require.NoError(t, err)
+		require.Equal(t, "invited", v.State)
 	})
 
-	t.Run("test success", func(t *testing.T) {
+	t.Run("test get connection record not found scenario", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte)}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+		valueFound, err := record.GetConnectionRecord("wrong-key")
+		require.Equal(t, err, storage.ErrDataNotFound)
+		require.Nil(t, valueFound)
+	})
+	t.Run("test get connection record empty key scenario", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte)}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+		valueFound, err := record.GetConnectionRecord("")
+		require.Contains(t, err.Error(), "input connectionID is empty")
+		require.Nil(t, valueFound)
+	})
+	t.Run("test get connection record ", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte)}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+		valueFound, err := record.GetConnectionRecord(connIDKeyPrefix + "_")
+		require.Equal(t, err, storage.ErrDataNotFound)
+		require.Nil(t, valueFound)
+	})
+}
+func TestConnectionRecorder_SaveConnectionRecord(t *testing.T) {
+	t.Run("save connection record success", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte)}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+		connRec := &ConnectionRecord{ThreadID: "xyz",
+			ConnectionID: "connID", State: stateNameInvited}
+		err := record.saveConnectionRecord(connRec)
+		require.NoError(t, err)
+	})
+	t.Run("save connection record error", func(t *testing.T) {
 		store := &mockstorage.MockStore{Store: make(map[string][]byte), ErrGet: fmt.Errorf("get error")}
 		record := NewConnectionRecorder(store)
 		require.NotNil(t, record)
-		require.NoError(t, store.Put("key1", []byte("value1")))
-		_, err := record.GetConnection("key1")
-		require.Error(t, err)
-		require.Contains(t, err.Error(), "get error")
+		connRec := &ConnectionRecord{ThreadID: "",
+			ConnectionID: "", State: stateNameInvited}
+		err := record.saveConnectionRecord(connRec)
+		require.Contains(t, err.Error(), "input connectionID is empty")
+	})
+}
+func TestConnectionRecorder_SaveProtocolID(t *testing.T) {
+	t.Run("save protocol ID and get connection id record", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte)}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+		request := &Request{
+			Type:  ConnectionRequest,
+			ID:    "requestID",
+			Label: "Bob",
+			Connection: &Connection{
+				DID: "B.did@B:A",
+			},
+		}
+		connRec := ConnectionRecord{ThreadID: "xyz",
```

Comment:
use constants, easy to read

Reply:
Is these constants are for did ? threadID etc